src/widgets/tab_viewer.rs

```diff
@@ -76,4 +76,14 @@ pub trait TabViewer {
     fn clear_background(&self, _tab: &Self::Tab) -> bool {
         true
     }
+    /// The scrolling of each tab is controlled by themselves
+
+    fn tab_scroll(&self, _tab: &Self::Tab) -> bool {
+        false
+    }
+    /// Is the tab's scrolling mode both, horizontal, or
```

Comment:
Move this one line down. I'd also rewrite this comment to "Determines if the horizontal and vertical scroll bars are shown for \`tab\`."

Reply:
Of course, I removed the qscrollarea in_tabs(), scroll_area_in_tabs parameter and tab scroll(), and I renamed scroll_both() to scroll_bars()！